src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -70,6 +72,77 @@ public void SettingUpdatesProperties()
             });
         }
 
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.Browser)]
+        public void SettingUpdatesPropertiesAfterAnother()
+        {
+            T item = GetExistingItem();
+
+            // These linq calls make an IEnumerable of pairs of functions that are not identical
+            // (eg. not (creationtime, creationtime)), includes both orders as seperate entries
+            // as they it have different behavior in reverse order (of functions), in addition
+            // to the pairs of functions, there is a reverse bool that allows a test for both
+            // increasing and decreasing timestamps as to not limit the test unnecessarily.
+            // Only testing with utc because it would be hard to check if lastwrite utc was the
+            // same type of method as lastwrite local since their .Getter fields are different.
+            var timeFunctionsUtc = TimeFunctions(requiresRoundtripping: true).Where((f) => f.Kind == DateTimeKind.Utc);
+            var booleanArray = new bool[] { false, true };
+            Assert.All(timeFunctionsUtc.SelectMany((x) => timeFunctionsUtc.SelectMany((y) => booleanArray.Select((reverse) => (x, y, reverse)))).Where((fs) => fs.x.Getter != fs.y.Getter), (functions) =>
+            {
+                // Checking that milliseconds are not dropped after setter.
+                // Emscripten drops milliseconds in Browser
+                DateTime dt1 = new DateTime(2002, 12, 1, 12, 3, 3, LowTemporalResolution ? 0 : 321, DateTimeKind.Utc);
+                DateTime dt2 = new DateTime(2001, 12, 1, 12, 3, 3, LowTemporalResolution ? 0 : 321, DateTimeKind.Utc);
+                DateTime dt3 = new DateTime(2000, 12, 1, 12, 3, 3, LowTemporalResolution ? 0 : 321, DateTimeKind.Utc);
+                if (functions.reverse)
+                {
+                    var swap = dt3;
+                    dt3 = dt1;
+                    dt1 = swap;
+                }
+                functions.x.Setter(item, dt1);
+                functions.y.Setter(item, dt2);
+                functions.x.Setter(item, dt3);
+                DateTime result_x = functions.x.Getter(item);
+                DateTime result_y = functions.y.Getter(item);
+                Assert.Equal(dt3, result_x);
+                Assert.Equal(dt2, result_y);
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(~(TestPlatforms.Browser | TestPlatforms.Windows))]
+        public void SettingUpdatesPropertiesOnSymlink()
+        {
+            // This test makes sure that the times are set on the symlink itself.
+            // It is needed as on OSX for example, the default for most APIs is
+            // to follow the symlink to completion and set the time on that entry
+            // instead (eg. the setattrlist will do this without the flag set).
+            T item = CreateSymlinkToItem(GetExistingItem());
+
+            Assert.All(TimeFunctions(requiresRoundtripping: true), (function) =>
```

Comment:
Hi, if we're going to do it on this simple method, shouldn't we also do it on the other preexisting simple tests such as `SettingUpdatesProperties`

Reply:
Hi, I've implemented locally changes that would allow for this to happen for almost every instance of TimeFunctions (based on commit https://github.com/dotnet/runtime/pull/49555/commits/19cbb7f815099b25c991ab249ef9b1e3b621275c). Please check it out and lmk if I should do it for all/some/none of these.
[patch.txt](https://github.com/dotnet/runtime/files/6496224/patch.txt)